File: HealthSPAR-main/product-webapp/healthspar-app/src/app/model/patient.ts
export interface Patient {
  patientId: string;
  patientName: string;
  email: string;
  phoneNumber: string;
  dob: Date;
  bloodGroup: string;
  gender: string;
  cityName: string;
  district: string;
  state: string;
  country: string;
  zip: string;
  medicalHistory: string;
  medicineHistory: string;
  treatmentHistory: string;
  fileName?: string;
  fileType?: string;
  imageData?: Uint8Array;
}

File: HealthSPAR-main/patient-service/src/main/java/com/stackroute/model/Patient.java
package com.stackroute.model;

import lombok.*;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.MongoId;

import java.time.LocalDate;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Document(collection = "patient_details")
public class Patient {
    @MongoId
    private String patientId;

    private String patientName;
    private String email;
    private String phoneNumber;
    private LocalDate dob;
    private String bloodGroup;
    private String gender;
    private String cityName;
    private String district;
    private String state;
    private String country;
    private String zip;
    private String medicalHistory;
    private String medicineHistory;
    private String treatmentHistory;

    private String fileName;
    private String fileType;
    private byte[] imageData;
}
